src/main/java/me/zeroeightsix/kami/module/modules/player/ViewLock.kt

```diff
@@ -0,0 +1,117 @@
+    private val yawSlice = register(Settings.integerBuilder("YawSlice").withValue(8).withRange(2, 32).withStep(1).withVisibility { autoYaw.value })
+    private val pitchSlice = register(Settings.integerBuilder("PitchSlice").withValue(5).withRange(2, 32).withStep(1).withVisibility { autoPitch.value })
+
+    private var yawSanp = 0
+    private var pitchSnap = 0
+    private val deltaXQueue = LinkedList<Pair<Int, Long>>()
+    private val deltaYQueue = LinkedList<Pair<Int, Long>>()
+    private var pitchSliceAngle = 1.0f
+    private var yawSliceAngle = 1.0f
+
+    override fun onEnable() {
+        yawSliceAngle = 360.0f / yawSlice.value
+        pitchSliceAngle = 180.0f / (pitchSlice.value - 1)
+        if ((autoYaw.value || autoPitch.value) && mc.player != null)
+            snapToNext()
+    }
+
+    init {
+        listener<SafeTickEvent> {
+            if (autoYaw.value || autoPitch.value) {
+                snapToSlice()
+            }
+            if (yaw.value && !autoYaw.value) mc.player.rotationYaw = specifcYaw.value
+            if (pitch.value && !autoPitch.value) mc.player.rotationPitch = specificPitch.value
+        }
+    }
+
+    private fun snapToNext() {
+        yawSanp = round(mc.player.rotationYaw / yawSliceAngle).toInt()
+        pitchSnap = round((mc.player.rotationPitch + 90) / pitchSliceAngle).toInt()
+        snapToSlice()
+    }
+
+    private fun changeDirection(yawChange: Int, pitchChange: Int) {
+        yawSanp = Math.floorMod(yawSanp + yawChange, yawSlice.value)
+        pitchSnap = (pitchSnap + pitchChange).coerceIn(0, pitchSlice.value - 1)
+        snapToSlice()
+    }
+
+    private fun snapToSlice() {
+        if (yaw.value && autoYaw.value) {
+            mc.player.rotationYaw = (yawSanp * yawSliceAngle).coerceIn(0f, 360f)
+            mc.player.ridingEntity?.let { it.rotationYaw = mc.player.rotationYaw }
+        }
+        if (pitch.value && autoPitch.value) mc.player.rotationPitch = (pitchSnap * pitchSliceAngle - 90).coerceIn(-90f, 90f)
+    }
+
+
+    fun handleDeltaX(deltaX: Int) {
+        val currenttime = System.currentTimeMillis()
+        deltaXQueue.add(Pair(deltaX, currenttime))
+        val sum = deltaXQueue.sumBy { it.first }
+        if (abs(sum) > threshold.value * 100) {
+            deltaXQueue.clear()
+            changeDirection(sign(sum.toDouble()).toInt(), 0)
+            return
+        }
+        while (deltaXQueue.peek().second < currenttime - 500)
+            deltaXQueue.remove()
+    }
+
+    fun handleDeltaY(deltaY: Int) {
+        val currenttime = System.currentTimeMillis()
+        deltaYQueue.add(Pair(deltaY, currenttime))
+        val sum = deltaYQueue.sumBy { it.first }
+        if (abs(sum) > threshold.value * 100) {
+            deltaYQueue.clear()
+            changeDirection(0, -sign(sum.toDouble()).toInt())
+            return
+        }
+        while (deltaYQueue.peek().second < currenttime - 500)
+            deltaYQueue.remove()
```

Comment:
I'll fix the lack of newlines myself, but use curly braces on the while please

Reply:
Doesn't matter if it is one liner

Reply:
Yes it does, I explained in another comment

Reply:
are braceless ifs fors and whiles never wanted ? ill make sure to not use them in future pr

Reply:
Either use inline or with braces please. In this specific case, braces would be better.